conans/client/conan_api.py

```diff
@@ -728,8 +728,9 @@ def search_packages(self, reference, query=None, remote_name=None, outdated=Fals
         return recorder.get_info()
 
     @api_method
-    def upload(self, pattern, package=None, remote_name=None, all_packages=False, force=False,
-               confirm=False, retry=2, retry_wait=5, skip_upload=False, integrity_check=False,
+    def upload(self, pattern, package=None, query=None, remote_name=None,
```

Comment:
I would say add the new argument in the last position, less likely to break to users of the API